### PR TITLE
New: markdown `rawContent()` and `compiledContent()` helpers

### PR DESCRIPTION
**Pull request by bholmesdev**
#### What kind of changes does this PR include?

- [ ] Minor content fixes (broken links, typos, etc.)
- [X] New or updated content
- [ ] Translated content
- [ ] Changes to the docs site code
- [ ] Something else!

#### Description

- We have all-new helpers for getting raw markdown content! This updates the imported markdown docs to explain how they work + common usage examples
- Add `rawContent()` explainer with "minutes read" example
- Add `rawContent.html()` explainer with "summary" example
- For discussion: **should we ditch the bulleted format and use level 4 headers now?** The examples are still readable, but not super scannable anymore

image

### CODE REVIEW

**Review by aFuzzyBear — APPROVED** [review]
I am really happy to see this feature brought back @bholmesdev 💚 
I would also second the move to h4 headings for the properties that are exported.
I think we should get this merged, and then formatting make the changes to the page in a separate PR

**Comment by bholmesdev** [comment]
Nice, exactly what I hoped for @sarah11918! I'll wait to merge until this new feature is _actually_ approved and published 😆

**Comment by sarah11918** [comment]
Our standard procedure for this @bholmesdev , is to mark the PR as draft if the feature/fix it relates to isn't merged/approved yet... juuuuuuust to be safe.  ;)

I'll do that now, and it's on you to come back and un-draft it and "update branch." But, no need let us know at that point. Just go ahead at your leisure!

File: src/pages/en/guides/markdown-content.md [hunk 1]
```diff
@@ -222,6 +222,42 @@ Each Markdown file exports the following properties:
 - `file`: The absolute path of this file (e.g. `/home/user/projects/.../file.md`).
 - `url`: If it's a page, URL of the page (e.g. `/en/guides/markdown-content`).
 - `getHeaders()`: An async function that returns the headers of the Markdown file. The response follows this type: `{ depth: number; slug: string; text: string }[]`.
+- `rawContent()`: A function that returns the raw content of the markdown file (excluding the frontmatter block) as a string. This is helpful when, say, calculating "minutes read." This example uses the [popular reading-time package](https://www.npmjs.com/package/reading-time):
```
Comment (Chrissdroid) [inline src/pages/en/guides/markdown-content.md:225]:
```suggestion
- `rawContent()`: A function that returns the raw content of the Markdown file (excluding the frontmatter block) as a string. This is helpful when, say, calculating "minutes read." This example uses the [popular reading-time package](https://www.npmjs.com/package/reading-time):
```

File: src/pages/en/guides/markdown-content.md [hunk 2]
```diff
@@ -222,6 +222,42 @@ Each Markdown file exports the following properties:
...
+
+  {posts.map((post) => (
+    <Fragment>
+      <h2>{post.frontmatter.title}</h2>
+      <p>{readingTime(post.rawContent()).text}</p>
+    </Fragment>
+  ))}
+  ```
+
+- `rawContent.html()`: An asynchronous function that returns the raw content parsed to HTML. Note: **this does not parse `{jsx expressions}`, `<Components />` or layouts**! Only standard markdown blocks like `## headings` and `- lists` will be parsed. This is useful when, say, rendering a summary block for a blog post. This example plucks out the first paragraph as a summary using the [popular node-html-parser package](https://www.npmjs.com/package/node-html-parser):
```
Comment (Chrissdroid) [inline src/pages/en/guides/markdown-content.md:241]: Maybe on it's own line ? And avoiding the repetition ("This is helpful when, say, ###" with "This is useful when, say, ###") from line 225
```suggestion
- `rawContent.html()`: An asynchronous function that returns the raw content parsed to HTML.

> **Note: This does not parse `{jsx expressions}`, `<Components />` or layouts**! Only standard Markdown blocks like `## headings` and `- lists` will be parsed. This is useful when rendering a summary block for a blog post. This example plucks out the first paragraph as a summary using the [popular node-html-parser package](https://www.npmjs.com/package/node-html-parser):
```

**Review by Chrissdroid — COMMENTED** [review]
Small suggestions to the file (like uppercase "Markdown" for consistency) or line return for better reading

File: src/pages/en/guides/markdown-content.md (hunk 2 above)
Comment (sarah11918) [reply src/pages/en/guides/markdown-content.md:241]: @Chrissdroid I'm totally with you, I didn't like the "squishy-ness" either! 

BUT, with bullet points at the moment, it's probably best just to leave it, and not have multiple paragraphs and call outs within bullet points. 

I have made an issue that we're going to go through this entire page and re-format/space things out to not use bullet points anymore, and we can fix it at that time! So, I told Ben it was fine as is.  :)

(Good spot on the Markdown above, too! 🚀  )

Comment (Chrissdroid) [reply src/pages/en/guides/markdown-content.md:241]: Updated suggestion ! 🚀 
```suggestion
- `rawContent.html()`: An asynchronous function that returns the raw content parsed to HTML. Note: **This does not parse `{jsx expressions}`, `<Components />` or layouts**! Only standard Markdown blocks like `## headings` and `- lists` will be parsed. This is useful when rendering a summary block for a blog post. This example plucks out the first paragraph as a summary using the [popular node-html-parser package](https://www.npmjs.com/package/node-html-parser):
```

Comment (sarah11918) [reply src/pages/en/guides/markdown-content.md:241]: Agreed! These are fan-nit-tastic notes for @bholmesdev !

Comment (Chrissdroid) [reply src/pages/en/guides/markdown-content.md]: I think we've agreed naturally on the docs on keeping uppercase format for libs and languages names like `React`, `Vue`, `Solid` or `JavaScript`, `TypeScript`, `Markdown`, etc...

**Pull request merged**